### ISSUE TITLE
'Usage' and 'Used in Process' features - for Elementary Flows - cannot see connections to processes housed in libraries

### ISSUE THREAD

**Issue by bnjmnmorelli** [GreenDelta/olca-app#420]
I have a database with some foreground modeling that is connecting to library versions of the USLCI and eLCI databases. It does not seem like the 'Usage' option (available when you right click) on an Elementary flow can see into libraries to tell you what processes are using that flow. The feature is able to identify connections between Elementary flows housed in a library and foreground processes and LCIA methods that use those flows. This issue is also affecting the list of processes that appear when you open an elementary flow and expand the 'Used in process' menu. These issues do not seem to be affecting technosphere flows.

**Comment by msrocka** [comment]
True, it is because the exchanges of processes and characterization factors of impact categories that are located in libraries are not stored in the database but in external matrices for faster loading in calculations. Only product outputs and waste inputs are stored in the database so that these can be linked in product systems. The usage search just scans the database (so it is also affecting the technosphere flows in case of product inputs and waste outputs). We would need to scan the matrices in the usage search too, but this needs some thinking. One of the main aspects of the usage search is to detect if something can be deleted, but this is not a problem here because it should not be possible to delete library flows (via the user interface).

**Comment by bnjmnmorelli** [comment]
Thanks Michael - This is helpful info. For some of our curation activities on the US Federal LCA Commons it would be nice to have the ability to search within libraries as we have the ability (and sometimes desire) to update databases that are converted to libraries. And it is sometimes more convenient/necessary to do these searches in the library itself. We can always go back and search the original database that was converted to a library, but it is an extra step that can be cumbersome. Perhaps this is something to consider if libraries become widely used. Thanks Again.
